### PR TITLE
Feat/review changes

### PR DESCRIPTION
**Pull request by raghul1212**

### CODE REVIEW

File: MYSQL scripts/interviewmanagementhibernate.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
+use sample;
```
Comment (abySebastin95) [inline MYSQL scripts/interviewmanagementhibernate.sql]: Using DB name is not advisable.

File: MYSQL scripts/interviewmanagementhibernate.sql [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ DELIMITER $;
+    CREATE PROCEDURE getCandidateUpdateByEmail(in email varchar(30),in id bigint)
+         BEGIN
+            SELECT * FROM candidate c where c.email_id=email and c.id!=id;
+         END 
+ $;
+#stored procedure for candidate entity starts
+#--------------------------------------------------------------------------------
+#Employee
+use sample;
```
Comment (abySebastin95) [inline MYSQL scripts/interviewmanagementhibernate.sql]: Using the DB name is not advisable.

File: MYSQL scripts/interviewmanagementhibernate.sql [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ $;
+
+DELIMITER $
+    CREATE PROCEDURE getCandidateByEmail(in email varchar(30) )
+         BEGIN
+            SELECT * FROM candidate c where c.email_id=email;
+         END 
+ $;
+ DELIMITER $;
+    CREATE PROCEDURE getCandidateUpdateByEmail(in email varchar(30),in id bigint)
```
Comment (abySebastin95) [inline MYSQL scripts/interviewmanagementhibernate.sql]: in id bigint - id means ?? Not meaning full

File: MYSQL scripts/interviewmanagementhibernate.sql (hunk 1 above)
Comment (abySebastin95) [inline MYSQL scripts/interviewmanagementhibernate.sql]: Log files -
${basePath}/application.log - This should not be pushed to github.

File: src/main/java/com/revature/interviewmanagement/dao/CandidateDao.java [hunk 4]
```diff
@@ -3,19 +3,18 @@
 import java.util.List;
 
 import com.revature.interviewmanagement.entity.Candidate;
+import com.revature.interviewmanagement.model.CandidateDto;
 
 public interface CandidateDao {
 
-	public String addCandidate(Candidate candidate);
+	public String addCandidate(Long id,CandidateDto candidate);
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/CandidateDao.java]: We can have the ID inside the DTO itself, why in separate params!!

File: src/main/java/com/revature/interviewmanagement/dao/EmployeeCredentialDao.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.revature.interviewmanagement.dao;
+
+import com.revature.interviewmanagement.model.credentials.EmployeeCredentialDto;
+
+public interface EmployeeCredentialDao {
+
+	public String updatePassword(Long id, EmployeeCredentialDto employeeCredential);
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/EmployeeCredentialDao.java]: Why separate param, if we are passing the whole DTO. !!

File: src/main/java/com/revature/interviewmanagement/dao/EmployeeDao.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+	public Employee getEmployeeByEmailId(String email);
+
+	public Employee getEmployeeByPhoneNumber(String phoneNumber);
+
+	public List<Employee> getEmployeeByDesignationId(Long destId);
+
+	public List<Employee> getEmployeeByEmployeeId(Long empId);
+
+	public String addEmployee(Long id,EmployeeDto employee);
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/EmployeeDao.java]: Why separate param, if we are passing the whole dto.

File: src/main/java/com/revature/interviewmanagement/dao/RecoveryPasswordDao.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.revature.interviewmanagement.dao;
+
+import com.revature.interviewmanagement.model.credentials.RecoveryPasswordDto;
+
+public interface RecoveryPasswordDao {
+
+	public void addCode(RecoveryPasswordDto recoverPassword, String forgotPasswordCode);
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/RecoveryPasswordDao.java]: Why separate param, if we are passing the whole dto.

File: src/main/java/com/revature/interviewmanagement/dao/ResultDao.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
+package com.revature.interviewmanagement.dao;
+
+import java.util.List;
+
+import com.revature.interviewmanagement.entity.Result;
+import com.revature.interviewmanagement.model.ResultDto;
+
+public interface ResultDao {
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/ResultDao.java:7]: By Default Interface has PUBLIC ACCESS MODIFIERS. We don't need to mention it here.

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+import com.revature.interviewmanagement.entity.credentials.CandidateCredential;
+import com.revature.interviewmanagement.exception.DuplicateIdException;
+import com.revature.interviewmanagement.exception.IdNotFoundException;
+import com.revature.interviewmanagement.model.credentials.CandidateCredentialDto;
+import com.revature.interviewmanagement.util.mapper.CandidateCredentialMapper;
+
+@Repository
+public class CandidateCredentialDaoImpl implements CandidateCredentialDao {
+
+	private static final Logger logger=LogManager.getLogger(CandidateCredentialDaoImpl.class.getName());
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: .get name is not necessary. 
ex: 
    private static final Logger logger = LogManager.getLogger(RevProServiceImp.class);

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+	
+	private static final String CHECK_EMAIL="SELECT c FROM CandidateCredential c WHERE c.emailId=:email";
+	private static final String CHECK_EMAIL_VALIDATE="SELECT c FROM CandidateCredential c WHERE c.emailId=:email";
+	
+	
+	@Override
+	public Boolean validateEmail(String email) {
+		Session session=sessionFactory.getCurrentSession();
+		logger.info("Entering validateEmailCredential method");
+		@SuppressWarnings("unchecked")
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: Not advisable to use - Suppress warnings.
Need to be checked and fixed.

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+	@Override
+	public CandidateCredential validateCredential(CandidateCredentialDto candidateCredentialDto) {
+		Session session=sessionFactory.getCurrentSession();
+		
+		logger.info("Entering validateCredential method");
+		@SuppressWarnings("unchecked")
+		List<CandidateCredential> resultList=session.createQuery(CHECK_EMAIL_VALIDATE)
+		.setParameter(EMAIL,candidateCredentialDto.getEmailId())
+		.getResultList();
+		return (resultList.isEmpty()?null:resultList.get(0));
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: get(0) !!

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	@Transactional
+	@Override
+	public String addCredential(CandidateCredentialDto candidateCredentialDto) {
+		Session session=sessionFactory.getCurrentSession();
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: This line is getting duplicated !!

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+		logger.info("Entering addCredential method");
+		@SuppressWarnings("unchecked")
+		List<CandidateCredential> resultList=session.createQuery(CHECK_EMAIL)
+				.setParameter(EMAIL,candidateCredentialDto.getEmailId())
+				.getResultList();
+		if(resultList !=null && resultList.isEmpty()) {
+			CandidateCredential candidateCredential=CandidateCredentialMapper.candidateCredentialMapper(candidateCredentialDto);
+			Long id=(Long) session.save(candidateCredential);
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: Type casting is not advisable.

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+			return "Candidate credentials created successfully with id "+id+" !";
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: return "Candidate credentials created successfully with id "+id+" !";
All these msg's can be moved to one Properties or Constant File - Common constant file.

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+		}
+		else {
+			throw new DuplicateIdException("Entered Email id already exists in another record");
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: Msg to Constant file.

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+				candidateCredential.setId(id);
+				candidateCredential.setAddedOn(updateObj.getAddedOn());
+				candidateCredential.setUpdatedOn(updateObj.getUpdatedOn());
+				candidateCredential.setUpdatedBy(updateObj.getUpdatedBy());
+				session.merge(candidateCredential);
+				session.flush();
+				}
+					
+		} 
+		catch (org.hibernate.ObjectNotFoundException e1) {
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: Not the right way to catch the exp.

File: src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+				.getResultList();
+		CandidateCredential candidateCredential=CandidateCredentialMapper.candidateCredentialMapper(candidateCredentialDto);
+		candidateCredential.setId(resultList.get(0).getId());
+		candidateCredential.setAddedOn(resultList.get(0).getAddedOn());
+		candidateCredential.setUpdatedOn(resultList.get(0).getUpdatedOn());
+		candidateCredential.setUpdatedBy(resultList.get(0).getUpdatedBy());
+		session.merge(candidateCredential);
+		session.flush();
+
+		return !resultList.isEmpty();
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/dao/impl/CandidateCredentialDaoImpl.java]: Not right way to return as response.

File: src/main/java/com/revature/interviewmanagement/entity/Result.java [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+	private LocalDateTime addedOn;
+	
+	@Column(name="updated_on")
+	private LocalDateTime updatedOn;
+	
+	@Column(name="updated_by")
+	private String updatedBy;
+	
+	@OneToOne
+	@JoinColumn(nullable=false)
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/entity/Result.java]: @JoinColumn(nullable=false)
 purpose of nullable  ??!!

File: src/main/java/com/revature/interviewmanagement/entity/credentials/CandidateCredential.java [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+	
+	@Column(name="updated_by")
+	private String updatedBy;
+	
+	@JsonIgnore
+	@OneToOne(mappedBy="candidateCredential")
+	private Candidate candidate;
+	
+	@JsonIgnore
+	@OneToOne(mappedBy="candidateCredential",cascade=CascadeType.ALL)
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/entity/credentials/CandidateCredential.java]: why cascadetype is all

File: src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+import com.revature.interviewmanagement.model.credentials.RecoveryPasswordDto;
+import com.revature.interviewmanagement.service.CandidateCredentialService;
+import com.revature.interviewmanagement.service.RecoveryPasswordService;
+import com.revature.interviewmanagement.utils.MailSenderUtil;
+import com.revature.interviewmanagement.utils.PasswordHasherUtil;
+import com.revature.interviewmanagement.utils.RandomCodeGeneratorUtil;
+
+
+@RestController
+@RequestMapping("/api")
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java]: / api is not the right word here.

File: src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+@CrossOrigin("http://localhost:4200")
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java]: this has to be domain-specific or *

File: src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java [hunk 22]
```diff
@@ -0,0 +1,147 @@
...
+			 sendCodeUtil(recoveryPasswordDto.getEmailId());
+			 recoveryPasswordService.addCode(recoveryPasswordDto,forgotPasswordCode);
+		 }
+		 
+		 return new ResponseEntity<>(result, new HttpHeaders(), HttpStatus.OK);  
+	}
+	 
+	
+	 
+	 @PutMapping("/candidate/credential/resend-code")
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java]: resend-code is not advisable - Not more than one word in mapping.

File: src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java [hunk 23]
```diff
@@ -0,0 +1,147 @@
...
+		 return new ResponseEntity<>(result, new HttpHeaders(), HttpStatus.OK);//true can be replaced with better approach
+	 }
+	 
+	 @PostMapping("/candidate/credential/check-code")
+	 public ResponseEntity<Boolean> validateCode(@RequestBody RecoveryPasswordDto recoveryPasswordDto){
+		 
+		 return new ResponseEntity<>(recoveryPasswordService.validateCode(recoveryPasswordDto), new HttpHeaders(), HttpStatus.OK); 
+	 }
+	 
+	 @PutMapping("/candidate/credential/reset-password")
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/rest/CandidateCredentialController.java]: /candidate can be moved to request mapping in the Top. 
/api should be removed.

File: src/main/java/com/revature/interviewmanagement/rest/ResultController.java [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+	
+	@DeleteMapping("/result/{id}")
+	public ResponseEntity<String> deleteResult(@PathVariable Long id){
+		logger.debug("Entering deleteResult method");
+		return	new ResponseEntity<>(resultService.deleteResult(id), new HttpHeaders(), HttpStatus.OK);
+	}
+	
+
+ 
+ @ExceptionHandler(IdNotFoundException.class)
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/rest/ResultController.java]: The controller should not have these.

File: src/main/java/com/revature/interviewmanagement/service/impl/CandidateCredentialServiceImpl.java [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+import com.revature.interviewmanagement.dao.CandidateCredentialDao;
+import com.revature.interviewmanagement.entity.credentials.CandidateCredential;
+import com.revature.interviewmanagement.model.credentials.CandidateCredentialDto;
+import com.revature.interviewmanagement.service.CandidateCredentialService;
+
+@Service
+public class CandidateCredentialServiceImpl implements CandidateCredentialService {
+
+	@Autowired
+	private CandidateCredentialDao candidateCredentialDao ;
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/service/impl/CandidateCredentialServiceImpl.java]: Extra space in between ;

File: src/main/java/com/revature/interviewmanagement/service/impl/InterviewServiceImpl.java [hunk 26]
```diff
@@ -0,0 +1,138 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.revature.interviewmanagement.dao.InterviewDao;
+import com.revature.interviewmanagement.entity.Interview;
+import com.revature.interviewmanagement.model.InterviewDto;
+import com.revature.interviewmanagement.service.InterviewService;
+
+@Service
```
Comment (abySebastin95) [inline src/main/java/com/revature/interviewmanagement/service/impl/InterviewServiceImpl.java:32]: Why do impl has @Service

**Review by abySebastin95 — CHANGES_REQUESTED** [review]

File: MYSQL scripts/interviewmanagementhibernate.sql (hunk 3 above)
Comment (raghul1212) [reply MYSQL scripts/interviewmanagementhibernate.sql]: Long datatype in entity is mapped as bigint in database

File: src/main/java/com/revature/interviewmanagement/dao/EmployeeDao.java (hunk 6 above)
Comment (raghul1212) [reply src/main/java/com/revature/interviewmanagement/dao/EmployeeDao.java]: Here, id refers to credential id of the employee which was created earlier and will be mapped to employee details entity.

**Pull request merged**